lib/screens/v2/screen_data.ex

```diff
@@ -54,7 +55,7 @@ defmodule Screens.V2.ScreenData do
   defp select_variant(screen_id, opts, then_fn) do
     config = get_config(screen_id, opts)
     selected_variant = Keyword.get(opts, :generator_variant)
-    query_params = Keyword.get(opts, :query_params)
+    query_params = Keyword.get(opts, :query_params) || %QueryParams{}
```

Comment:
There is a `Keyword.get/3` which accepts a default value if the key isn't found, and I think that'd be more natural than `||` here.

Reply:
of course there is, Elixir is great I should've assumed, will update